packages/core/src/App/Containers/AccountSwitcher/account-switcher.jsx

```diff
@@ -22,121 +22,121 @@ import { AccountsItemLoader } from 'App/Components/Layout/Header/Components/Prel
+    React.useEffect(() => {
+        document.addEventListener('mousedown', handleClickOutside);
+        props.updateMt5LoginList();
+        return () => {
+            props.toggleShouldShowRealAccountsList(false);
+            document.removeEventListener('mousedown', handleClickOutside);
+        };
+    }, [handleClickOutside]);
+
+    const handleLogout = () => {
+        closeAccountsDialog();
+        if (props.is_positions_drawer_on) {
+            props.togglePositionsDrawer(); // TODO: hide drawer inside logout, once it is a mobx action
         }
-        this.props.logoutClient().then(() => {
-            this.props.routeBackInApp(this.props.history);
+        props.logoutClient().then(() => {
+            props.routeBackInApp(props.history);
         });
     };
 
-    closeAccountsDialog = () => {
-        this.props.toggleAccountsDialog(false);
+    const closeAccountsDialog = () => {
+        props.toggleAccountsDialog(false);
     };
 
-    redirectToMt5 = account_type => {
-        this.closeAccountsDialog();
-        this.props.history.push(`${routes.mt5}#${account_type}`);
+    const redirectToMt5 = account_type => {
+        closeAccountsDialog();
+        props.history.push(`${routes.mt5}#${account_type}`);
     };
 
-    openMt5RealAccount = account_type => {
+    const openMt5RealAccount = account_type => {
         const has_required_account =
-            account_type === 'synthetic' ? this.props.has_malta_account : this.props.has_maltainvest_account;
-
-        if (this.props.is_eu && !has_required_account) {
-            this.closeAccountsDialog();
-            this.props.openAccountNeededModal(
-                account_type === 'synthetic'
-                    ? this.props.standpoint.gaming_company
-                    : this.props.standpoint.financial_company,
+            account_type === 'synthetic' ? props.has_malta_account : props.has_maltainvest_account;
+
+        if (props.is_eu && !has_required_account) {
+            closeAccountsDialog();
+            props.openAccountNeededModal(
+                account_type === 'synthetic' ? props.standpoint.gaming_company : props.standpoint.financial_company,
                 account_type === 'synthetic' ? localize('Deriv Synthetic') : localize('Deriv Financial'),
                 account_type === 'synthetic' ? localize('DMT5 Synthetic') : localize('DMT5 Financial')
             );
         } else {
             sessionStorage.setItem('open_mt5_account_type', `real.${account_type}`);
-            this.redirectToMt5Real();
+            redirectToMt5Real();
         }
     };
 
-    redirectToMt5Real = () => {
-        if (!this.props.is_logged_in || this.props.is_mt5_allowed) {
-            this.redirectToMt5('real');
+    const redirectToMt5Real = () => {
+        if (!props.is_logged_in || props.is_mt5_allowed) {
+            redirectToMt5('real');
         } else {
             window.open(urlFor('user/metatrader', { legacy: true }));
         }
```

Comment:
I know this is out of context, but can we remove this else block here? 

@negarn Would you please confirm this?

Reply:
@reza-fs we should refactor the usage of `is_mt5_allowed` in multiple places from my short investigation.

Reply:
I've addressed this here https://github.com/binary-com/deriv-app/pull/2289